### PR TITLE
Backport `pbench-move-results --controller`

### PR DESCRIPTION
**Pull request by portante**
This is a back-port of PR #1990, which also fixes the unit tests in the `util-scripts` directory to check for dangling tests, and also correctly test expected status.

### CODE REVIEW

**Review by portante — COMMENTED** [review]
Ready for review.

**Review by ndokos — APPROVED** [review]
LGTM

**Comment by ndokos** [comment]
PR #1990 also saves the output of `hostname -i`, something which this PR does not do. Should it?

**Comment by portante** [comment]
> PR #1990 also saves the output of `hostname -i`, something which this PR does not do. Should it?

I felt that should be a new feature for 0.71 and was not worth back-porting.

**Review by webbnh — APPROVED** [review]
Looks OK to me.  See below for the usual collection of nits 'n' bits for your consideration, as well as a couple of questions.

File: agent/util-scripts/pbench-make-result-tb [hunk 1]
```diff
@@ -84,7 +84,7 @@ while true; do
 	    break;
 	    ;;
 	*)
- 	    printf "\n%s: you specified an invalid option or an unexpected argument: \"${opt}\"\n\n" "${script_name}" >&2    
+	    printf "\n%s: you specified an invalid option or an unexpected argument: \"${opt}\"\n\n" "${script_name}" >&2
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/pbench-make-result-tb:87]: We are all over the place on hard tabs vs soft tabs vs indentation levels.

In this case, the back-port was done by "copy from `main` to `b0.69` and get working," which picked up the removal of the unwanted leading space.

File: agent/util-scripts/pbench-make-result-tb [hunk 2]
```diff
@@ -132,10 +134,22 @@ fi
 
 res_name=$(getconf.py --config ${mdlog} name pbench)
 if [[ "${res_name}" != "${pbench_run_name}" ]]; then
-    warn_log "The run in directory ${pbench_run}/${pbench_run_name} has an unexpected metadata name, \"${res_name}\" - skipping"
+    error_log "The run in directory ${pbench_run}/${pbench_run_name} has an unexpected metadata name, \"${res_name}\" - skipping"
     exit 1
 fi
 
+printf -- "%s" "${full_hostname}" | pbench-add-metalog-option ${mdlog} pbench hostname_f
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/pbench-make-result-tb:141]: Existing code, not changing.

File: agent/util-scripts/pbench-make-result-tb [hunk 3]
```diff
@@ -102,6 +102,8 @@ if [[ ! -d "${target_dir}" ]]; then
     exit 1
 fi
 
+controller_dir="$(basename -- "${target_dir}")"
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/pbench-make-result-tb:105]: The same as they do in `main`.  Nothing. =)

File: agent/util-scripts/pbench-move-results [hunk 4]
```diff
@@ -72,19 +80,39 @@ while true; do
     esac
 done
 
-if [[ ! -d "${pbench_run}" ]]; then
+if [[ -z "${controller}" ]]; then
+    controller=${full_hostname}
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/pbench-move-results:84]: Agreed.  Something that will be taken care of on the `main` branch when we re-write in Python3.  =)

File: agent/util-scripts/pbench-move-results [hunk 5]
```diff
@@ -72,19 +80,39 @@ while true; do
...
+trap "rm -rf ${tmp}" EXIT INT QUIT TERM
+
+# Before we take any further actions, enumerate the results to be moved
+mkdir ${tmp}
+sts=${?}
+if [[ ${sts} -ne 0 ]] ;then
+    error_log "Failed: \"mkdir -p ${tmp}\", status ${sts}"
     exit 1
 fi
+/bin/ls -ort -d */ | awk '{print $8}' | grep -v "^tools-" | grep -v "^tmp/" > ${tmp}/results.lis
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/pbench-move-results:108]: Agreed, but did not change the `ls` command from the previous code, and we are re-writing this in Python3 on the `main` branch.

File: agent/util-scripts/unittests [hunk 6]
```diff
@@ -199,16 +199,18 @@ function _verify_output {
...
+        if [[ ${sts} -eq ${expected_status} ]]; then
+            if [[ ${expected_status} -ne 0 ]]; then
+                echo "PASS - ${tname} exited with expected status: ${sts}"
+                sts=0
+            else
+                echo "PASS - ${tname}"
+            fi
         else
-            echo "FAIL - ${tname}: PASS output but execution returned non-zero exit status"
+            echo "FAIL - ${tname}: PASS output but execution returned unexpected exit status: ${sts}"
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/util-scripts/unittests:210]: Tracked in #2284.

**Pull request merged**